EIPS/eip-3290.md

```diff
@@ -0,0 +1,172 @@
+
+3. An insurance fund
+Lock funds and tokens temporarily in a smart contract as a way of insuring them until a certain date.
+
+4. Pension or Savings account
+Another application of locking funds could be to create a small pension fund or time-based savings account, i.e., one that prevents the owner from withdrawing any funds before a certain time in the future. (It could be particularly useful for addicted crypto traders in helping keep their ether intact.)
+
+5. Family trust fund
+Let's imagine we would like to give ether, ERC token, or NFT to someone for their 18th birthday. We could write down on a piece of paper the beneficiary account's private key and the address holding the funds and hand it over to them in an envelope. The only thing they would have to do is call a function on the contract from their account once they are 18, and all the funds will be transferred to them.
+
+6. Vesting
+The most common reason at the moment to lock funds is called "vesting." Imagine that you have just raised a successful ICO and your company still holds a majority of tokens distributed between your team members.
+It would be beneficial to all parties involved to ensure that people's tokens cannot be traded straightaway. If there are no controls in place, anyone might take action by selling all their tokens, cashing out, and quitting the company. This would negatively affect the market price and make all remaining contributors to the project unhappy.
+
+7. DEFI Yielding
+Protect the deposited liquidity or the pool itself.
+
+## Specification
+TBS (Time Based Switch) feature integrates smart contracts and uses events/callbacks to inform the user about the TBS state.
+
+The user creates his TBS by calling the createSwitch function together with four parameters. This returns a signed transaction and puts it in a queue for due execution based on block height or a timestamp.
+
+```
+function createSwitch(uint _time, uint _amount, address_executor, address payable _benefitor)
+    doesntExist
+    checkAmount(_amount)
+    checkTime(_time)
+    public
+    Payable 
+{...}
+```
+
+TBS creation will produce a signed transaction that is put on hold until/if the trigger hits.
+All signed transactions are stored in the TBS queue object in the contract.
+
+Only the executor address is able to trigger a funds transfer when the defined timestamp has expired. The triggered event can be defined as any type of transaction or a specific transaction or blockchain information. Besides that, the creator will determine a percentage of balance dispersed per fallback address.
+
+``` 
+function tryExecuteSwitch(address account)
+    onlyValid(account)
+    onlyExecutorsOrOwner(account)
+    public
+    Payable
+{...}
+```
+
+Required event listener methods for tokens transfer are:
+```approveToken, withdrawToken```
```

Comment:
If these are Ethereum events, they should contain a full definition.

Reply:
Could you please elaborate on what do you mean by full definition?

Reply:
```solidity
event approveToken(address token, uint256 amount);
```